awx/api/serializers.py

```diff
@@ -1333,6 +1333,8 @@ def validate(self, attrs):
             scm_type = attrs.get('scm_type', u'') or u''
         if self.instance and not scm_type:
             valid_local_paths.append(self.instance.local_path)
+        if scm_type and self.instance.local_path != attrs['local_path']:
+            errors['local_path'] = _('Cannot change local_path for projects with SCM')
```

Comment:
Could we make this error message a bit more explicit, i.e..,

`_(f'Cannot change local_path for {scm_type}-based projects')`